### PR TITLE
Add stretch kinematics plugin

### PR DESCRIPTION
**Pull request by JafarAbdi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# stretch_kinematics_plugin
+
+Description: MoveIt 2 kinematics plugin for stretch robot from Hello Robot Inc.
+
+### Setup
+
+To setup the kinematics plugin we need a joint model group that consist of two joint model groups one for the manipulator and the other one for the mobile base which have only one joint with planar type so the `*.srdf` file should have
```
Comment (mamoll) [inline README.md]:
```suggestion
To setup the kinematics plugin we need a joint model group that consist of two joint model groups: one for the manipulator and the other one for the mobile base which has only one joint with planar type so the `*.srdf` file should have
```

File: src/stretch_kinematics_plugin.cpp [hunk 2]
```diff
@@ -0,0 +1,313 @@
+#include <stretch_kinematics_plugin/stretch_kinematics_plugin.h>
+
+#include <tf2/transform_datatypes.h>
+#include <tf2_eigen/tf2_eigen.h>
+
+namespace stretch_kinematics_plugin
+{
+static rclcpp::Logger LOGGER = rclcpp::get_logger("stretch_kinematics_plugin");
+
+StretchKinematicsPlugin::StretchKinematicsPlugin() : initialized_(false)
```
Comment (mamoll) [inline src/stretch_kinematics_plugin.cpp]: Is this considered more idiomatic member initialization than using `bool initialized_{false}` in the class declaration? I am just curious what is considered good practice.

File: src/stretch_kinematics_plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    state_->updateLinkTransforms();
+    // The transformation for the tip frame w.r.t. the mobile base link given the arm's joint values
+    const auto mobile_base_link_T_current =
+        state_->getGlobalLinkTransform(mobile_base_joint_->getChildLinkModel()).inverse() *
+        state_->getGlobalLinkTransform(getTipFrame());
+    // The error for the mobile base link
+    const auto T_mobile_base_link = eigen_ik_pose * mobile_base_link_T_current.inverse();
+
+    Eigen::AngleAxisd rotation(T_mobile_base_link.rotation());
+    Eigen::Map<Eigen::Vector2d>(&solution.at(mobile_base_index_)) = T_mobile_base_link.translation().block<2, 1>(0, 0);
```
Comment (mamoll) [inline src/stretch_kinematics_plugin.cpp:272]: it looks like you set the x,y position of the base always to 0,0. is that right?

File: src/stretch_kinematics_plugin.cpp [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+    // First we solve for the arm only by fixing the mobile base
+    state_->setJointGroupPositions(mobile_base_jmg_, &solution[mobile_base_index_]);
+    state_->updateLinkTransforms();
+    // Convert the desired pose from the arm + mobile base solver base frame to the arm group base frame
+    const auto arm_b_T_ik_pose = tf2::toMsg(state_->getGlobalLinkTransform(arm_ik_solver->getBaseFrame()).inverse() *
+                                            state_->getGlobalLinkTransform(getBaseFrame()) * eigen_ik_pose);
+    const bool ik_valid = arm_ik_solver->searchPositionIK(arm_b_T_ik_pose, arm_jmg_ik_seed_state, timeout * 0.25,
+                                                          arm_jmg_consistency_limits, arm_jmg_solution,
+                                                          solution_callback, error_code, options);
+    if (ik_valid || options.return_approximate_solution)  // found acceptable solution
```
Comment (mamoll) [inline src/stretch_kinematics_plugin.cpp:248]: even if options.return_approximate_solution is true, shouldn't it still update the ik solution for the base before returning?

**Review by mamoll — APPROVED** [review]
I have some questions, but overall this looks good!

Would this work if the arm is modeled with a bunch of mimic joints?

File: src/stretch_kinematics_plugin.cpp (hunk 2 above)
Comment (JafarAbdi) [reply src/stretch_kinematics_plugin.cpp]: Yes, that's right

File: src/stretch_kinematics_plugin.cpp (hunk 3 above)
Comment (JafarAbdi) [reply src/stretch_kinematics_plugin.cpp:272]: No, I'm setting it to the first two rows of  `T_mobile_base_link.translation()` which are the x,y part of the translation, this's equivalent to `T_mobile_base_link.translation()[0:2, 0:1]` in numpy

**Pull request merged**